File: Hanseungwoo/Week2/problem3.py
# https://www.acmicpc.net/problem/11279

import heapq

n = int(input()) 

h = []

for _ in range(n):
    x = int(input())  

    if x == 0:
        if len(h) == 0:
            print(0)
        else:
            num = heapq.heappop(h)
            print(-num)
    else:
        heapq.heappush(h, -x)

File: Kwonsaebom/Week3/problem3.js
// https://www.acmicpc.net/problem/2839

const fs = require("fs");
const input = Number(fs.readFileSync("/dev/stdin").toString().trim());

let total = Number(input);

const big = 5;
const small = 3;

let count = 0;

while (total >= 0) {
  if (total % 5 === 0) {
    // 0일때도 포함됨
    count += total / 5;
    console.log(count);
    return;
  }
  total -= 3;
  count++;
}
console.log(-1);
